### PR TITLE
python3Packages.python-poppler: init at 0.4.1

### PR DESCRIPTION
**Pull request by onny**
Adding Python module [python-poppler](https://github.com/cbrunet/python-poppler), „Python binding to poppler-cpp“.

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by fabaff** [comment]
## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 351620`

---
### `x86_64-linux`
<details>
  <summary>:white_check_mark: 4 packages built:</summary>
  <ul>
    <li>python311Packages.python-poppler</li>
    <li>python311Packages.python-poppler.dist</li>
    <li>python312Packages.python-poppler</li>
    <li>python312Packages.python-poppler.dist</li>
  </ul>
</details>

File: pkgs/development/python-modules/python-poppler/default.nix [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+
+  checkInputs = [ pytestCheckHook ];
+
+  pythonImportCheck = [ "poppler" ];
+
+  meta = {
+    description = "Python binding to poppler-cpp";
+    homepage = "https://github.com/cbrunet/python-poppler";
+    changelog = "https://cbrunet.net/python-poppler/changelog.html";
+    license = lib.licenses.gpl2Only;
```
Comment (fabaff) [inline pkgs/development/python-modules/python-poppler/default.nix:62]: According to the file headers (e.g., https://github.com/cbrunet/python-poppler/blob/master/src/poppler/__init__.py) it looks like `GPLv2+`.

```suggestion
    license = lib.licenses.gpl2Plus;
```

Comment (onny) [reply pkgs/development/python-modules/python-poppler/default.nix]: Thank you, fixed it!

File: pkgs/development/python-modules/python-poppler/default.nix [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+  };
+
+  patches = [
+    # Prevent Meson from downloading pybind11, use system version instead
+    ./use_system_pybind11.patch
+  ];
+
+  build-system = [ setuptools ];
+
+  build-inputs = [ pybind11 ];
```
[the remark this answers is not in the document]
Comment (onny) [reply pkgs/development/python-modules/python-poppler/default.nix]: Ups, my fault, fixed it!

File: pkgs/development/python-modules/python-poppler/default.nix [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+    ninja
+    pkg-config
+  ];
+
+  dependencies = [
+    meson-python
+    poppler
+  ];
+
+  checkInputs = [ pytestCheckHook ];
```
Comment (natsukium) [inline pkgs/development/python-modules/python-poppler/default.nix:49]:
```suggestion
  nativeCheckInputs = [ pytestCheckHook ];
```

File: pkgs/development/python-modules/python-poppler/default.nix [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+    pname = "python_poppler";
+    hash = "sha256-5spcI+wCNQvyzvhaa/nxsmF5ZDbbR4F2+dJPsU7uzGo=";
+  };
+
+  patches = [
+    # Prevent Meson from downloading pybind11, use system version instead
+    ./use_system_pybind11.patch
+  ];
+
+  build-system = [ setuptools ];
```
Comment (natsukium) [inline pkgs/development/python-modules/python-poppler/default.nix:34]:
```suggestion
  build-system = [ meson-python ];
```
https://github.com/cbrunet/python-poppler/blob/7bfcf8243e0162dba87791e928a644efac807a27/pyproject.toml#L1-L3

File: pkgs/development/python-modules/python-poppler/default.nix [hunk 5]
```diff
@@ -0,0 +1,62 @@
...
+  buildInputs = [ pybind11 ];
+
+  nativeBuildInputs = [
+    meson
+    ninja
+    pkg-config
+  ];
+
+  dependencies = [
+    meson-python
```
Comment (natsukium) [inline pkgs/development/python-modules/python-poppler/default.nix:45]:
```suggestion
```

File: pkgs/development/python-modules/python-poppler/default.nix [hunk 6]
```diff
@@ -0,0 +1,62 @@
...
+    poppler
```
Comment (natsukium) [inline pkgs/development/python-modules/python-poppler/default.nix:46]: `poppler` is not a python dependency, so it should go to `propagatedBuildInputs`.

**Review by natsukium — DISMISSED** [review]

**Comment by onny** [comment]
Really good to know, thank you @natsukium, fixed it!

**Comment by natsukium** [comment]
Result of `nixpkgs-review pr 351620` at aa8854e1 run on aarch64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>:white_check_mark: 2 packages built:</summary>
  <ul>
    <li>python311Packages.python-poppler</li>
    <li>python312Packages.python-poppler</li>
  </ul>
</details>

**Review by natsukium — APPROVED** [review]
LGTM

**Pull request merged**